### PR TITLE
Merge SkipFailingEventsTest with EngineErrorHandlingTest

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

~:x: Blocked by https://github.com/camunda/zeebe/pull/10888~

Merge SkipFailingEventsTest with EngineErrorHandlingTest, to make more clear what they actually testing
and in which responsibility this test lies.

## Related issues

related #10455 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by Zelldon** [comment]
@oleschoenburg I rebased the PR please have a look :)

File: engine/src/test/java/io/camunda/zeebe/engine/processing/streamprocessor/EngineErrorHandlingTest.java [hunk 1]
```diff
@@ -40,7 +75,10 @@
 public final class EngineErrorHandlingTest {
 
   private static final String STREAM_NAME = "foo";
+  private static final ProcessInstanceRecord PROCESS_INSTANCE_RECORD = Records.processInstance(1);
+  private static final JobRecord JOB_RECORD = Records.job(1);
 
+  protected SynchronousLogStream stream;
```
Comment (oleschoenburg) [inline engine/src/test/java/io/camunda/zeebe/engine/processing/streamprocessor/EngineErrorHandlingTest.java:81]: :wrench: can be removed

File: engine/src/test/java/io/camunda/zeebe/engine/processing/streamprocessor/EngineErrorHandlingTest.java [hunk 2]
```diff
@@ -40,7 +75,10 @@
 public final class EngineErrorHandlingTest {
 
   private static final String STREAM_NAME = "foo";
+  private static final ProcessInstanceRecord PROCESS_INSTANCE_RECORD = Records.processInstance(1);
```
Comment (oleschoenburg) [inline engine/src/test/java/io/camunda/zeebe/engine/processing/streamprocessor/EngineErrorHandlingTest.java:78]: :wrench: I'd not define it here and just inline everywhere. It's not a complex statement, doesn't save much resources and actually inlining it makes some tests easier to follow. For example, I was confused about `shouldBlacklistInstance` until I realized that `PROCESS_INSTANCE_RECORD` is a process instance with key 1

File: engine/src/test/java/io/camunda/zeebe/engine/processing/streamprocessor/EngineErrorHandlingTest.java [hunk 3]
```diff
@@ -40,7 +75,10 @@
...
+  private static final JobRecord JOB_RECORD = Records.job(1);
```
Comment (oleschoenburg) [inline engine/src/test/java/io/camunda/zeebe/engine/processing/streamprocessor/EngineErrorHandlingTest.java:79]: :wrench: Same comment as above for `PROCESS_INSTANCE_RECORD`

**Review by lenaschoenburg — APPROVED** [review]
LGTM :+1:

File: engine/src/test/java/io/camunda/zeebe/engine/processing/streamprocessor/EngineErrorHandlingTest.java (hunk 1 above)
Comment (Zelldon) [reply engine/src/test/java/io/camunda/zeebe/engine/processing/streamprocessor/EngineErrorHandlingTest.java:81]: Argh seem to come back due to resolving merge conflicts sorry.

**Comment by Zelldon** [comment]
Thanks @oleschoenburg good points :) I applied the requested changes.

bors r+

**Pull request merged**